Letterbook.Web/Program.cs

```diff
@@ -0,0 +1,84 @@
+using Letterbook.Adapter.ActivityPub;
+using Letterbook.Api;
+using Letterbook.Api.Swagger;
+using Letterbook.Core;
+using Letterbook.Core.Exceptions;
+using Letterbook.Core.Extensions;
+using Serilog;
+using Serilog.Enrichers.Span;
+using Serilog.Events;
+
+namespace Letterbook.Web;
+
+public class Program
+{
+	public static void Main(string[] args)
+	{
+		// Pre initialize Serilog
+		Log.Logger = new LoggerConfiguration()
+			.MinimumLevel.Override("Microsoft", LogEventLevel.Information)
+			.Enrich.FromLogContext()
+			.Enrich.WithSpan()
+			.WriteTo.Console()
+			.CreateBootstrapLogger();
+
+		var builder = WebApplication.CreateBuilder(args);
+		var coreOptions = builder.Configuration.GetSection(CoreOptions.ConfigKey).Get<CoreOptions>()
+		                  ?? throw new ConfigException(nameof(CoreOptions));
+
+		if (!builder.Environment.IsProduction())
+			builder.Configuration.AddUserSecrets<Api.Program>();
+		// Register Serilog - Serialized Logging (configured in appsettings.json)
+		builder.Host.UseSerilog((context, services, configuration) => configuration
+			.Enrich.FromLogContext()
+			.Enrich.WithSpan()
+			.ReadFrom.Configuration(context.Configuration)
+			.ReadFrom.Services(services)
+		);
+
+		builder.Services.AddApiProperties(builder.Configuration);
+		builder.Services.AddTelemetry();
+		builder.Services.AddHealthChecks();
+		builder.Services.AddActivityPubClient(builder.Configuration);
+		builder.Services.AddServices(builder.Configuration);
+		builder.Services.AddRazorPages();
+
+		builder.WebHost.UseUrls(coreOptions.BaseUri().ToString());
+
+		var app = builder.Build();
+
+		// Configure the HTTP request pipeline.
+		if (!app.Environment.IsDevelopment())
+		{
+			// Not sure if this works, with mixed Razor/WebApi
+			app.UseExceptionHandler("/Error");
+		}
+
+		if (!app.Environment.IsProduction())
+		{
+			app.Use((context, next) =>
+			{
+				context.Request.EnableBuffering();
+				return next();
+			});
+			app.UseSwaggerConfig();
+		}
+
+		app.UseStaticFiles();
+
+		app.UseHealthChecks("/healthz");
+		app.MapPrometheusScrapingEndpoint();
+		app.UseWebFingerScoped();
+
+		app.UseAuthentication();
+		app.UseAuthorization();
+
+		app.UseSerilogRequestLogging();
+
+		app.MapRazorPages();
+		app.UsePathBase(new PathString("/api/v1"));
+		app.MapControllers();
```

Comment:
How do we intend to connect the frontend and backend together?

Do we expect this server will proxy calls to a separately hosted Letterbook.API server?

Reply:
The web project actually just hosts the api endpoints, alongside the razor pages. That's what `MapControllers()` is doing.

I expect that some views can use built in razor forms and otherwise just call the API from javascript.